huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_instance.go

```diff
@@ -1111,3 +1144,74 @@ func switchAuditLog(client *golangsdk.ServiceClient, instanceId string, v bool)
 
 	return nil
 }
+
+func switchSQLFilter(d *schema.ResourceData, client *golangsdk.ServiceClient, instanceId string,
+	isSQLFilterEnabled bool) error {
+	flag := "OFF"
+	if isSQLFilterEnabled {
+		flag = "ON"
+	}
+	opts := sqlfilter.UpdateSqlFilterOpts{
+		SwitchStatus: flag,
+	}
+
+	res, err := sqlfilter.Update(client, instanceId, opts).ExtractJobResponse()
+	if err != nil {
+		return fmt.Errorf("switch SQL filter to %q failed: %s", flag, err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Pending", "Running", "Failed"},
+		Target:       []string{"Completed"},
+		Refresh:      gaussDBMysqlDatabaseStatusRefreshFunc(client, instanceId, res.JobID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        1 * time.Second,
+		PollInterval: 1 * time.Second,
+	}
+
+	//nolint:staticcheck
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for switch SQL filter job (%s) to complete: %s", instanceId, err)
+	}
+	return nil
+}
+
+func gaussDBMysqlDatabaseStatusRefreshFunc(client *golangsdk.ServiceClient, instanceId,
+	jobId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		var (
+			getJobStatusHttpUrl = "v3/{project_id}/jobs"
+		)
+
+		getJobStatusPath := client.Endpoint + getJobStatusHttpUrl + buildQueryJobParam(jobId)
+		getJobStatusPath = strings.ReplaceAll(getJobStatusPath, "{project_id}", client.ProjectID)
+		getJobStatusPath = strings.ReplaceAll(getJobStatusPath, "{instance_id}", instanceId)
```

Comment:
it seems that the getJobStatusHttpUrl  does not include `{instance_id}`

Reply:
done